config/src/lib.rs

```diff
@@ -45,7 +43,9 @@ trait Configurable {
     ///   and it was present on the command line
     /// - Some(Vec<String>) with one or more list elements: parameter that takes
     ///   a value and one or more values were specified
-    fn parse_values(parsed_values: HashMap<ConfigOption, Option<Vec<String>>>) -> Self;
+    fn parse_values(
+        parsed_values: HashMap<ConfigOption, Option<Vec<String>>>,
+    ) -> Result<Box<Self>, anyhow::Error>;
```

Comment:
I don't think you need any of these Box things as long as Configurable is Sized. I'm not 100% sure but removing all of those and running the tests works.

I also think that anyhow shouldn't appear in the API (but I'm not sure about that tbh.)

Reply:
I think trying to return the trait Error from these functions in the result was what got me started down the Box<..> route, so using anyhow::error removed the need for Boxes all over the place. I agree that it would be nicer to define a proper error type and have those error be part of the api though.